src/lib.rs

```diff
@@ -210,18 +209,20 @@ impl AstarteSdk {
     /// ```no_run
     /// #[tokio::main]
     /// async fn main() {
-    ///     let mut sdk_options = astarte_sdk::builder::AstarteOptions::new("_","_","_","_")
+    ///     let mut sdk_options = astarte_device_sdk::builder::AstarteOptions::new("_","_","_","_")
     ///                           .build();
-    ///     let mut d = astarte_sdk::AstarteSdk::new(&sdk_options).await.unwrap();
+    ///     let mut d = astarte_device_sdk::AstarteDeviceSdk::new(&sdk_options).await.unwrap();
     ///
     ///     loop {
-    ///         if let Ok(data) = d.poll().await {
+    ///         if let Ok(data) = d.astarte_device_data_event().await {
     ///             println!("incoming: {:?}", data);
     ///         }
     ///     }
     /// }
     /// ```
-    pub async fn poll(&mut self) -> Result<Clientbound, AstarteError> {
+    pub async fn astarte_device_data_event(
```

Comment:
Maybe `astarte_device_event` is better because it also receives the `connack` and `publish` event. What do you think @Annopaolo ?

Reply:
Even if `connack` and `publish` are actually MQTT events, they play an important role in the Astarte MQTTv1 protocol. So I think it's correct to keep the `astarte_` prefix. I agree with your 'astarte_device_event` suggestion. Maybe also `astarte_device_event_handler` , but I'm not sure it conforms to Rust naming best practices.

Reply:
I don't see any issue with the `astarte_device_event_handler` naming. I was also wondering about it myself.